### PR TITLE
Milestone 1 - OOP school library: Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by grauJavier**
### ✅ Completed Project Requirements
- The `Person` class has been created with the following features:
  - Instance variables: `@id`, `@name`, and `@age`.
  - A constructor that accepts `name`, `age`, and `parent_permission` as parameters. `name` and `parent_permission` have default values of `"Unknown"` and `true`.
  - Getters for `@id`, `@name`, and `@age`.
  - Setters for `@name` and `@age`.
  - A private method `of_age?` that returns `true` if `@age` is greater than or equal to 18, and `false` otherwise.
  - A public method `can_use_services?` that returns `true` if the person is of age or has parental permission.

- The `Student` class has been successfully created, inheriting from the `Person` class. It includes:
  - A constructor that extends the parent's constructor by adding `@classroom` and a corresponding parameter.
  - A method `play_hooky` that humorously returns `"¯\(ツ)/¯"`.

- The `Teacher` class has been implemented, also inheriting from the `Person` class. It features:
  - A constructor that extends the parent's constructor by adding `@specialization` and a corresponding parameter.
  - An override of the `can_use_services?` method so that it always returns `true`.

- Each class has been organized into its own separate file for better code structure.

### CODE REVIEW

File: person_class.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Person
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
```
Comment (OyePriscilla) [inline person_class.rb]: Kindly implement few features that are missing from this class as stated in the requirement: 👇 

1.  GETTERS AND SETTERS methods are not yet created for `@id, @name, and @age and setters for @name and @age`. You can do this by `using atrr_reader and attr_accessor`. ℹ️ 
2. `@id` that should be generated randomly is not yet implemented.
3. `def of_age?` is expected to be a private method.

File: person_class.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  def can_use_services?
+    if of_age? || @parent_permission
+      true
+    else
+      false
+    end
+  end
```
Comment (OyePriscilla) [inline person_class.rb]: - [x] In order to simpily your code base, the `can_use_services?` method is a Ruby method that returns a boolean value, the **`?`** at the end of the method name implies that it's a predicate method that either return `true or false`. ℹ️ 

```suggestion
 def can_use_services? 
    of_age?() || @parent_permission
  end
```

File: student_class.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person_class'
+
+class Student < Person
+  def
+    super(_name, _age, _parent_permission)
+    @classroom
+  end
```
Comment (OyePriscilla) [inline student_class.rb]: - [x] Kindly add the initialize method, the initialize method is a special method that gets called when an object is created from a class. It is commonly used to set up initial values for the instance variables of an object. 🙏 

```suggestion
 def initialize(age, classroom, name = "Unknown", parent_permission = true)`
    super(_name, _age, _parent_permission)
    @classroom = classroom
  end
```

File: teacher_class.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person_class'
+
+class Teacher < Person
+  def
+    super(_name, _age)
+    @specialization
+  end
```
Comment (OyePriscilla) [inline teacher_class.rb]: - [x] Kindly add the initialize method, the initialize method is a special method that gets called when an object is created from a class. It is commonly used to set up initial values for the instance variables of an object. 🙏

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @grauJavier ✔️ , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Implemented classes and objects in Ruby ✔️ 
Implemented encapsulation and inheritance with Ruby ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @grauJavier , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :fire: 
- All requested  changes implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**